016/default

payne-chris-r opened the pull request.

payne-chris-r committed:
Pulls in new files necessary for default

Pulls over new files from an `ember-new` app. Adds new file
structure like app/models, app/components, etc.

payne-chris-r committed:
Moves application template to proper place

Replaces `ember-new` application template with `ember-template`'s
previous one. Also deletes old application template

payne-chris-r committed:
Moves serializer and adapter to their own folders

Moves the serializer and adapter code to their respective
folders in order to abandon pod structure.

payne-chris-r committed:
Uses ember generator for tests and fix ember-cli

Ember-cli was previously broken (invalid JSON because extra `,`).
Fixed the cli, then recreated the adapter, serializer, and service
using ember generator, because they didn't have tests.

payne-chris-r commented:
Not sure if the auth service should be in there as blank, but I thought it'd give us a place to start...

raq929 commented on app/app.js:
```diff
@@ -10,7 +10,7 @@ Ember.MODEL_FACTORY_INJECTIONS = true;
 App = Ember.Application.extend({
   modulePrefix: config.modulePrefix,
   podModulePrefix: config.podModulePrefix,
-  Resolver,
```
Why are we deleting trailing commas in js?

payne-chris-r replied:
I think that's my mistake. js-standard doesn't like them, and I'm in that weird half-way stage of implementing some js-standard stuff, and not others.

raq929 commented:
There's a failing test on this PR having to do with the application serializer. The serializer doesn't actually DO anything, so I'm not sure how much it matters? But I think it should pass or be deleted.

raq929 commented:
I didn't mean to close, just comment.

raq929 commented:
Also, I think we should probably deal with these warnings, although I guess that's updating node globally.
```
DEPRECATION: Node v5.10.1 is no longer supported by Ember CLI. We recommend that you use the most-recent "Active LTS" version of Node.js.
WARNING: Node v5.10.1 is not tested against Ember CLI on your platform. We recommend that you use the most-recent "Active LTS" version of Node.js.
```

payne-chris-r commented:
Is this the one you're talking to?
```
Promise rejected before "it serializes records": Attempting to register an unknown factory: 'model:application'
```
?

raq929 commented:
That's the one.

raq929 commented:
There's no application model. So I think we can safely delete the application serializer and this test. Because I lied. On master, this test doesn't exist.

payne-chris-r commented:
It's possible that I didn't create the application serializer correctly. I'm gonna take a look.

payne-chris-r commented:
I feel like `application` shouldn't have a model, but maybe it's showing that there's an issue with how `active-model-serializer` is being brought in?

raq929 commented:
Right, the serializer expects a model. Otherwise what would it serialize? So I think we can kill it?

payne-chris-r commented:
Sounds like we're trying to extend an adapter that doesn't exist... I'm punting on this because it sounds like an issue with the version of ember-data we're using, not with the current update (see my soon-to-be-pushed branch). As you'll see my vote on this is to leave #23 open and continue doing what I was trying to do (de-podify), and come back to this later (if necessary) with 23.

payne-chris-r committed:
Replaces serializer test with previous version's

The old version of ember-template had a blank serializer test.
Issue #23 is open because of this, but isn't something we should
be resolving now. The focus of this merge is to de-podify, which
I am hopefully completing by moving the serializer test to the right
directory, and deleting the (now obsolete) tests from the pod-structure.
